clickplc/driver.py

```diff
@@ -698,6 +856,74 @@ async def _set_dh(self, start: int, data: list[int]):
             raise ValueError('Data list longer than available addresses.')
         await self.write_registers(address, values=data)
 
+    async def _set_yd(self, start: int, data: list[int]):
+        """Set YD registers. Called by `set`."""
+        # make sure the values are correct
+        #   side note: yd0u will come in with start == 0.5
+        if start < 0 or start > 8:
+            raise ValueError("YD must be in [0, 8]")
+        # make sure all the data is an int16
+        for datum in data:
+            if datum.bit_length() > 16:
+                raise ValueError(f"Datum {datum} is longer than 16 bits. YD registers cannot hold more than 16 bits.")
+        # get the correct starting address
+        address = int(57856 + 2 * (start))
+
+        # i am going to do this horribly. anyone who comes after me and wants
+        #   to fix it is absolutely welcome to.
+        horrible_index = self.u_index(start)
+        if len(data) > 10 - horrible_index:
+            raise ValueError(
+                "Data list is longer than available addresses. " +
+                "Make sure you're accounting for YD0u!"
+                )
+
+        # yd sucks. sorry i have to do it like this
+        values: list[int] = []
+        extended_zero = False
+        for i, datum in enumerate(data):
+            if (start == 0 and i in (0, 1)) or (start == 0.5 and i == 0):
+                values.append(datum)
+            else :
+                extended_zero = True
+                values.extend((datum, 0x0000))
+        # remove the last (0x0000)
+        if extended_zero:
+            values.pop()
+        await self.write_registers(address, values)
+        return
+
+    @staticmethod
+    def u_index(x: int | float) -> int:
+        """Here's the deal with this method.
+
+        I had to denote for XD and YD if somebody was trying to get/set
+        XD0u or YD0u. So if that happens, then I pass through 0.5 as the
```

Comment:
Hmm, yeah whoever came up with `XD0u` or `YD0u` ... fucked up.  And you can tell them I said so.

Rather than making this whole convoluted mess of _all the other XD/YD registers_, let's instead special-case `XD0u` and `YD0u` higher up so that `_set_xd` etc. never has to deal with it